apps/web/src/components/DashboarNotebookGroupButton.tsx

```diff
@@ -19,23 +24,32 @@ function DashboardNotebookGroupButton(props: Props) {
             ? 'bg-ceramic-50 text-gray-900 -mr-px'
             : 'bg-white text-gray-500'
         )}
-        href={`/workspaces/${props.workspaceId}/documents/${props.documentId}/notebook${props.isEditing ? '/edit' : ''}`}
+        href={`/workspaces/${props.workspaceId}/documents/${
+          props.documentId
+        }/notebook${props.isEditing ? '/edit' : ''}`}
       >
         <BookOpenIcon className="w-4 h-4" />
         <span>Notebook</span>
       </Link>
-      <Link
+      <button
         className={clsx(
           'flex gap-x-1.5 items-center rounded-sm px-3 py-1 text-sm disabled:cursor-not-allowed disabled:opacity-50 ring-1 ring-inset ring-gray-300 hover:bg-ceramic-50',
           props.current === 'dashboard'
             ? 'bg-ceramic-50 text-gray-900 -ml-px'
             : 'bg-white text-gray-500'
         )}
-        href={`/workspaces/${props.workspaceId}/documents/${props.documentId}/dashboard${props.isEditing ? '/edit' : ''}`}
+        disabled={props.userRole === 'viewer' && !props.isPublished}
+        onClick={() => {
+          router.push(
+            `/workspaces/${props.workspaceId}/documents/${
+              props.documentId
+            }/dashboard${props.isEditing ? '/edit' : ''}`
+          )
+        }}
       >
         <Squares2X2Icon className="w-4 h-4" />
         <span>Dashboard</span>
-      </Link>
+      </button>
```

Comment:
Can we add a tooltip here explaining why this is disabled?
Also, are we redirecting `viewers` if they land on `/dashboard`?